packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx

```diff
@@ -162,59 +51,14 @@ const EAHomeMainContent = ({FrontpageNode, classes}:{
   FrontpageNode: ComponentType,
   classes: ClassesType
 }) => {
-  // we use the widths of the tabs window and the underlying topics bar
-  // when calculating how far to scroll left and right
-  const tabsWindowRef = useRef<HTMLDivElement|null>(null)
-  const topicsBarRef = useRef<HTMLDivElement|null>(null)
-
-  // we store the topic bar scrollLeft offsets that correspond to displaying each "set" of topics
-  const offsets = useRef<Array<number>>([0])
-
-  useEffect(() => {
-    if (!tabsWindowRef.current || !topicsBarRef.current) return
-    offsets.current = [0]
-    Array.from(topicsBarRef.current.children).forEach((topic: HTMLElement) => {
-      if (!tabsWindowRef.current || !topicsBarRef.current) return
-      // we are looking for the topic that would get cut off at the end of each "set",
-      // by checking if the right edge would be past the window
-      // - if so, this will be the first in the next "set"
-      if (topic.offsetLeft + topic.offsetWidth - offsets.current[offsets.current.length-1] > tabsWindowRef.current.offsetWidth) {
-        // subtract 30px to account for the fade on the left side of the tabs window
-        offsets.current.push(topic.offsetLeft - 30)
-      }
-    })
-  }, [tabsWindowRef, topicsBarRef])
-  
-  const { results: coreTopics } = useMulti({
-    terms: {view: "coreTags"},
-    collectionName: "Tags",
-    fragmentName: 'TagDetailsFragment',
-    limit: 40
-  })
-  let allTabs: TopicsBarTab[] = useMemo(() => {
-    let tabs: TopicsBarTab[] = [frontpageTab]
-    if (coreTopics) {
-      // list the core topics based on topicTabsOrder
-      topicTabsOrder.forEach(topicId => {
-        const topic = coreTopics.find(t => t._id === topicId)
-        if (topic) {
-          tabs.push(topic)
-        }
-      })
-    }
-    return tabs
-  }, [coreTopics])
-  
+  const frontpageTab = useMemo(() => ({_id: '0', name: FRONTPAGE_TAB_NAME}), [])
```

Comment:
This doesn't need to be in the component right? Could we just leave it where it was in the code above?

Reply:
I think I needed it in the component part way through the refactoring or smth, but yeah moving it out would be simpler vs using useMemo